spyder/widgets/findreplace.py

```diff
@@ -444,10 +461,41 @@ def replace_find(self, focus_replace_text=False):
                                        QTextCursor.KeepAnchor)
                 else:
                     break
-                if not self.all_check.isChecked():
+                if not replace_all:
                     break
-            self.all_check.setCheckState(Qt.Unchecked)
             if cursor is not None:
                 cursor.endEditBlock()
             if focus_replace_text:
                 self.replace_text.setFocus()
+
+    @Slot()
+    def replace_find_all(self, focus_replace_text=False):
+        """Replace and find all matching occurrences"""
+        self.replace_find(focus_replace_text, replace_all=True)
+
+                
+    @Slot()
+    def replace_find_selection(self, focus_replace_text=False):

```

Comment:
I'm not sure if using a custom search is the best way, because the other searching methods use the [findText](http://doc.qt.io/qt-5/qwebengineview.html#findText) function of `QWebEngineView`.

Although I'm not sure if this could be implemented using `QWebEngineView.findText` because it doesn't support searching in a selection, also multiple selection should be managed adding more complexity.